app/controllers/hosts_controller.rb

```diff
@@ -47,6 +47,10 @@ def index(title = nil)
     end
     respond_to do |format|
       format.html do
+        if Setting[:new_hosts_page]
+          redirect_to(new_hosts_index_page_path)
+          next
+        end
```

Comment:
If you do this, why can't it be at at the very beginning? So line 42.

Reply:
Updated to try your approach. @jeremylenz what do you think ?